src/Symfony/Component/Security/Core/User/LdapUserProvider.php

```diff
@@ -35,8 +35,12 @@ class LdapUserProvider implements UserProviderInterface
     private $defaultSearch;
     private $passwordAttribute;
 
-    public function __construct(LdapInterface $ldap, string $baseDn, string $searchDn = null, string $searchPassword = null, array $defaultRoles = array(), string $uidKey = 'sAMAccountName', string $filter = '({uid_key}={username})', string $passwordAttribute = null)
+    public function __construct(LdapInterface $ldap, string $baseDn, string $searchDn = null, string $searchPassword = null, array $defaultRoles = array(), string $uidKey = 'sAMAccountName', ?string $filter = '({uid_key}={username})', string $passwordAttribute = null)
     {
+        if (null === $filter) {
+            $filter = '({uid_key}={username})';
+        }
```

Comment:
Is it still useful to have the default value in the argument list at this point? It's defined twice now with 2 different ways of making it a default value. I think the default value in the argument list can be removed. If I understand the need correctly, it's because you want to pass the `$passwordAttribute` while the `$filter` remains default, but it's an issue to write it out without constant?